gslib/tests/test_context_config.py

```diff
@@ -179,12 +183,36 @@ def testDoesNotProvisionIfUseClientCertificateNotTrue(self, mock_Popen):
     context_config.create_context_config(self.mock_logger)
     mock_Popen.assert_not_called()
 
-  def testRaisesErrorIfCertProviderCommandAbsent(self):
+  @mock.patch.object(subprocess, 'Popen')
+  @mock.patch(OPEN_TO_PATCH, new_callable=mock.mock_open)
+  def testDefaultCommand(self, mock_Popen, mock_open):
+    mock_open.return_value = DEFAULT_METADATA
     with SetBotoConfigForTest([('Credentials', 'use_client_certificate', 'True')
                               ]):
-      with self.assertRaises(context_config.CertProvisionError):
+      # Purposely end execution here to avoid writing a file.
+      with self.assertRaises(ValueError):
+        context_config.create_context_config(self.mock_logger)
+
+        mock_open.assert_called_once_with(
+            '~/.secureConnect/context_aware_metadata.json')
+        mock_Popen.assert_called_once_with(os.path.realpath('some/helper'),
+                                           '--print_certificate',
+                                           '--with_passphrase')
+
+  @mock.patch.object(subprocess, 'Popen')
+  def testCustomCommand(self, mock_Popen):
```

Comment:
The test naming format is "testSomethingDoesSomething".
For example, "testCustomCommandGeneratesCorrectCert".